### PR TITLE
docs(core): add missing docs to component and fix formatting

### PR DESCRIPTION
**Pull request by kara**
This should resolve the rest of the comments on https://github.com/angular/angular/pull/20855.

cc @jelbourn

### CODE REVIEW

**Review by jelbourn — APPROVED** [review]
LGTM

File: packages/core/src/render3/component.ts [hunk 1]
```diff
@@ -17,94 +19,122 @@ import {notImplemented, stringify} from './util';
...
- * Options which control how the component should be bootstrapped.
- */
-export interface CreateComponentOptionArgs {
-  /**
-   * Which renderer factory to use.
-   */
+/** Options that control how the component should be bootstrapped. */
+export interface CreateComponentOptions {
+  /** Which renderer to use. */
+  renderer?: Renderer3;
```
Comment (mhevery) [inline packages/core/src/render3/component.ts:25]: I think that is a bad merge. There is no `renderer` any more just `rendereFactory`

Comment (mhevery) [reply packages/core/src/render3/component.ts:25]: Fixed it for you.

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
merged as 9fd052032a

**Pull request closed without merging**